speech/check_dataVals.m

```diff
@@ -157,32 +180,73 @@ function reload_dataVals(src,evt)
     shortTrials = [];
     longTrials = [];
     nanFTrials = [];
-    jumpF1Trials = [];
-    jumpF2Trials = [];
-    fishyF1Trials = [];
+    for s = 1:UserData.nSigs
+        jumpTrials.(UserData.sigs2plot{s}) = [];
+    end
+    fishyTrials = [];
     earlyTrials = [];
     lateTrials = [];
     goodTrials = [];
 
     %% put trials into error categories
     for i = 1:length(dataVals)
+        %create vector of maximum differences, only for good trials and
+        %trials that are not just nans
+        maxDiffs = zeros(1,UserData.nSigs);
+        bNaNVals = all(isnan(dataVals(i).(UserData.sigs2plot{1})));
+        if strcmp(UserData.sigs2plot{s},'f0') %for f0 exclude first 40 ms of signal
+            %find trial with >1 sample
+            for j = 1:length(UserData.dataVals)
+                if length(UserData.dataVals(j).pitch_taxis) > 1
+                    tstep = diff(UserData.dataVals(j).pitch_taxis(1:2)); % get time step for each sample
+                    break;
+                end
+            end
```

Comment:
Is UserData.dataVals the same as dataVals? (this is already inside a dataVals loop) Maybe this is just to find the first trial with a valid pitch_taxis? But it seems like we don't have to do this for every trial, could just do once before the dataVals loop.

Reply:
We now get `tstep` outside the dataVals loop. 7ab80680e560364f54942fd2434f19fa64f287b3